core/src/main/scala/kafka/controller/ControllerState.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.controller
+
+import scala.collection.Seq
+
+sealed abstract class ControllerState {
+
+  def value: Byte
+
+  def rateAndTimeMetricName: Option[String] =
+    if (hasRateAndTimeMetric) Some(s"${toString}RateAndTimeMs") else None
+
+  protected def hasRateAndTimeMetric: Boolean = true
+}
+
+object ControllerState {
```

Comment:
The names below a little inconsistent although they match the KIP. More concretely: Idle, ControllerChange, BrokerChange, TopicChange, IsrChange, ControlledShutdown, TopicDeletion versus PartitionReassigning, AutoLeaderBalancing, ManualLeaderBalancing. It seems like we should do either of the following:

1. Change the former to be: Idling, ControllerChanging, BrokerChanging, etc.
2. Change the latter to be: PartitionReassignment, AutoLeaderBalance, ManualLeaderBalance

Thoughts?